docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -380,3 +383,96 @@ The Java code generated for this variant is:
     public Published(Long year, String publisher) { /* ... */ }
     public Record toValue() { /* ... */ }
   }
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
+The Java Code Generator uses Java Generic types to represent :ref:`DAML parameterized types <daml-ref-parameterized-types>`.
+
+Below is a DAML fragment defining the parameterized type ``Attribute`` for use by the ``BookAttribute`` type for modeling
+the characteristics of the book.
+
+.. literalinclude:: ./code-snippets/ParameterizedTypes.daml
+   :language: daml
+   :lines: 3-11
+   :caption: Com/Acme.daml
+
+A file Java file is generated for the ``Attribute`` data type that defines the Java Generic class:
+
+.. code-block:: java
+  :caption: com/acme/Attribute.java
+  :emphasize-lines: 3,8,10
+
+  package com.acme;
+
+  public class Attribute<a> {
+    public final a value;
+
+    public Attribute(a value) { /* ... */  }
+
+    public Record toValue(Function<a, Value> toValuea) { /* ... */ }
+
+    public static <a> Attribute<a> fromValue(Value value$, Function<Value, a> fromValuea) { /* ... */ }
+  }
+
+Serializing
+"""""""""""
+
+To serialize an instance of the ``Attribute<a>`` data type, a function for creating the Ledger API equivalent of the attribute value
+is passed to the ``toValue`` method as the ``fromValuea`` argument (see the above ``com/acme/Attribute.java`` source extract).
```

Comment:
```suggestion
is passed to the ``toValue`` method as the ``fromValue`` argument (see the above ``com/acme/Attribute.java`` source extract).
```

Reply:
Maybe somewhat surprisingly, this is not a typo but the generated parameter name `fromValue` for the type parameter `a` -> `fromValuea`. I agree, not very intuitive without an explanation.

Reply:
oh, my bad! An explanation would be great.

Reply:
Add explanation.